plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,273 @@
+            return;
+        }
+
+        SearchResponse response = queryTask(nodeId);
+
+        SearchHit[] hits = response.getHits().getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        if (hits.length != 1) {
+            log.error("Can't get the latest successful deployment in node id {}", nodeId);
+        }
+
+        int reTryTimes = 0;
+        try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hits[0].getSourceRef())) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            MLTask mlTask = MLTask.parse(parser);
+
+            autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+        } catch (RuntimeException | IOException e) {
+            reTryTimes++;
+            // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+            saveLatestReTryTimes(nodeId, reTryTimes);
+            log.error("Can't auto reload model in node id {} ,has try {} times\nThe reason is:{}", nodeId, reTryTimes, e.getMessage());
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED"
+     * and "worker_node" match nodeId
+     * @param nodeId one of query condition
+     * @return SearchResponse
+     */
+    @VisibleForTesting
+    SearchResponse queryTask(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
+            .must(QueryBuilders.matchPhraseQuery("worker_node", nodeId));
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder sortBuilderOrder = new FieldSortBuilder("last_update_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        searchRequest.source(searchSourceBuilder).indices(ML_TASK_INDEX);
+        try {
+            return client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+        } catch (IndexNotFoundException e) {
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
+        }
+    }
+
+    /**
+     *  auto reload 1 model under the node id
+     * @param nodeId node id
+     * @param modelId model id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
```

Comment:
Use all nodeIds fetched from task index, otherwise this can break other functionalities.

Reply:
but our aim is to reload model on 1 node halt down with unknown reason. so the other nodes may be run model normally. use all nodes to load, will corrupt the status of models on the other nodes.

Reply:
This won't corrupt the status of other node's models, and if not doing this, this will break the profile api functionality, because profile api reads ml task index to get the expected deployment plan.